docs/rag/ilab-rag-retrieval.md

```diff
@@ -0,0 +1,437 @@
+
+Options:
+  --document-store-type TEXT      The document store type, one of:
+                                  `milvuslite`.
+  --document-store-uri TEXT       The document store URI
+  --document-store-collection-name TEXT
+                                  The document store collection name
+  --model-dir TEXT                Base directories where models are stored.
+                                  [default: (The default system model location
+                                  store, located in the data directory.)]
+  --embedding-model TEXT          The embedding model name
+  --output-dir TEXT               Directory where generated datasets are
+                                  stored.
+  --input DIRECTORY               Directory where pre-processed documents are
+                                  located.
+  --help                          Show this message and exit.
+```
+
+| Option Description | Default Value | CLI Flag | Environment Variable |
+|--------------------|---------------|----------|----------------------|
+| Document store implementation, one of: `milvuslite`, **TBD** | `milvuslite` | `--document-store-type` | `ILAB_DOCUMENT_STORE_TYPE` |
+| Document store service URI. | `./embeddings.db` | `--document-store-uri` | `ILAB_DOCUMENT_STORE_URI` |
+| Document store collection name. | `IlabEmbeddings` | `--document-store-collection-name` | `ILAB_DOCUMENT_STORE_COLLECTION_NAME` |
+| Base directories where models are stored. | `$HOME/.cache/instructlab/models`  | `--retriever-embedder-model-dir` | `ILAB_EMBEDDER_MODEL_DIR` |
+| Name of the embedding model. | **TBD** | `--retriever-embedder-model-name` | `ILAB_EMBEDDER_MODEL_NAME` |
+
+### 2.6 RAG Chat Pipeline Command
+
+The proposal is to add a `chat.rag.enable` configuration (or the equivalent `--rag` flag) to the `model chat` command, like:
+
+```bash
+ilab model chat --rag
+```
+
+#### Command Purpose
+
+This command enhances the existing `ilab model chat` functionality by integrating contextual information retrieved from user-provided documents,
+enriching the conversational experience with relevant insights.
+
+#### Revised chat pipeline
+
+* Start with the user's input, `user_query`.
+* Use the given `user_query` to retrieve relevant contextual information from the embedding database (semantic search).
+* Append the retrieved context to the original LLM request.
+* Send the context augmented request to the LLM and return the response to the user.
+
+#### Prompt Template
+
+A default non-configurable template is used with parameters to specify the user query and the context, like:
+
+```text
+Given the following information, answer the question.
+Context:
+{context}
+Question: 
+{user_query}
+Answer:
+```
+
+Future extensions should align prompt management with the existing InstructLab design.
+
+### 2.7 RAG Chat Commands
+
+The `/r` command may be added to the `ilab model chat` command to dynamically toggle the execution of the RAG pipeline.
```

Comment:
Again, let's qualify this by saying that `/r` will only appear in this help list and only work if ILAB_DEV_PREVIEW is true.